ui/app/models/secret-v2.js

```diff
@@ -31,23 +31,40 @@ export default Model.extend(KeyMixin, Validations, {
   updatedTime: attr(),
   currentVersion: attr('number'),
   oldestVersion: attr('number'),
+  customMetadata: attr('object', {
+    editType: 'kv',
+    subText: 'An optional set of informational key-value pairs that will be stored with all secret versions.',
+  }),
   maxVersions: attr('number', {
     defaultValue: 10,
-    label: 'Maximum Number of Versions',
+    label: 'Maximum number of versions',
+    subText:
+      'The number of versions to keep per key. Once the number of keys exceeds the maximum number set here, the oldest version will be permanently deleted.',
   }),
   casRequired: attr('boolean', {
     defaultValue: false,
     label: 'Require Check and Set',
-    helpText:
-      'Writes will only be allowed if the key’s current version matches the version specified in the cas parameter',
+    subText:
+      'Writes will only be allowed if the key’s current version matches the version specified in the cas parameter.',
+  }),
+  deleteVersionAfter: attr({
+    defaultValue: 0,
+    editType: 'ttl',
+    label: 'Automate secret deletion',
+    helperTextDisabled: 'A secret’s version must be manually deleted.',
+    helperTextEnabled: 'Delete all new versions of this secret after',
   }),
   fields: computed(function() {
-    return expandAttributeMeta(this, ['maxVersions', 'casRequired']);
+    return expandAttributeMeta(this, ['customMetadata', 'maxVersions', 'casRequired', 'deleteVersionAfter']);
   }),
-  versionPath: lazyCapabilities(apiPath`${'engineId'}/data/${'id'}`, 'engineId', 'id'),
-  secretPath: lazyCapabilities(apiPath`${'engineId'}/metadata/${'id'}`, 'engineId', 'id'),
+  secretDataPath: lazyCapabilities(apiPath`${'engineId'}/data/${'id'}`, 'engineId', 'id'),
+  secretMetadataPath: lazyCapabilities(apiPath`${'engineId'}/metadata/${'id'}`, 'engineId', 'id'),
+
+  canListMetadata: alias('secretMetadataPath.canList'),
+  canReadMetadata: alias('secretMetadataPath.canRead'),
+  canUpdateMetadata: alias('secretMetadataPath.canUpdate'),
 
-  canEdit: alias('versionPath.canUpdate'),
```

Comment:
renamed to be clear whether we were talking about the data or metadata endpoint.